### PR TITLE
add support for meter/meterTestnet

### PR DESCRIPTION
**Pull request by simonzg**
## PR-Codex overview
This PR focuses on adding Meter chain and Meter Testnet chain. 

### Detailed summary
- Added Meter chain and Meter Testnet chain
- Exported `meter` and `meterTestnet` from `chains/definitions/index.ts`
- Created `meter.ts` and `meterTestnet.ts` files in `chains/definitions` directory
- Imported `defineChain` function from `utils/chain.js` in `meter.ts` and `meterTestnet.ts`
- Defined chain configurations for Meter and Meter Testnet chains

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/chains/index.ts [hunk 1]
```diff
@@ -131,3 +131,5 @@ export { zkSyncTestnet } from './definitions/zkSyncTestnet.js'
 export { zetachainAthensTestnet } from './definitions/zetachainAthensTestnet.js'
 export { zora } from './definitions/zora.js'
 export { zoraTestnet } from './definitions/zoraTestnet.js'
+export { meter } from './definitions/meter.js'
```
Comment (jxom) [inline src/chains/index.ts:134]: These need to be in alphabetical order.

File: src/chains/definitions/meterTestnet.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  rpcUrls: {
+    default: { http: ['https://rpctest.meter.io'] },
+    public: { http: ['https://rpctest.meter.io'] },
+  },
+  blockExplorers: {
+    etherscan: { name: 'MeterTestnetScan', url: 'https://scan-warringstakes.meter.io' },
+    default: { name: 'MeterTestnetScan', url: 'https://scan-warringstakes.meter.io' },
+  },
+  contracts: {
+  },
```
Comment (jxom) [inline src/chains/definitions/meterTestnet.ts:21]:
```suggestion
```

File: src/chains/definitions/meter.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  rpcUrls: {
+    default: { http: ['https://rpc.meter.io'] },
+    public: { http: ['https://rpc.meter.io'] },
+  },
+  blockExplorers: {
+    etherscan: { name: 'MeterScan', url: 'https://scan.meter.io' },
+    default: { name: 'MeterScan', url: 'https://scan.meter.io' },
+  },
+  contracts: {
+  },
```
Comment (jxom) [inline src/chains/definitions/meter.ts:21]:
```suggestion
```

**Review by jxom — CHANGES_REQUESTED** [review]

**Comment by simonzg** [comment]
suggested changes are included in the latest commit

File: src/chains/definitions/meter.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    default: { http: ['https://rpc.meter.io'] },
+    public: { http: ['https://rpc.meter.io'] },
+  },
+  blockExplorers: {
+    etherscan: { name: 'MeterScan', url: 'https://scan.meter.io' },
+    default: { name: 'MeterScan', url: 'https://scan.meter.io' },
+  },
+  contracts: {
+    
+  },
```
Comment (tmm) [inline src/chains/definitions/meter.ts:22]:
```suggestion
```

File: src/chains/definitions/meter.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    decimals: 18,
+    name: 'MTR',
+    symbol: 'MTR',
+  },
+  rpcUrls: {
+    default: { http: ['https://rpc.meter.io'] },
+    public: { http: ['https://rpc.meter.io'] },
+  },
+  blockExplorers: {
+    etherscan: { name: 'MeterScan', url: 'https://scan.meter.io' },
```
Comment (tmm) [inline src/chains/definitions/meter.ts:17]: Not a Etherscan explorer.

```suggestion
```

File: src/chains/definitions/meterTestnet.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    decimals: 18,
+    name: 'MTR',
+    symbol: 'MTR',
+  },
+  rpcUrls: {
+    default: { http: ['https://rpctest.meter.io'] },
+    public: { http: ['https://rpctest.meter.io'] },
+  },
+  blockExplorers: {
+    etherscan: { name: 'MeterTestnetScan', url: 'https://scan-warringstakes.meter.io' },
```
Comment (tmm) [inline src/chains/definitions/meterTestnet.ts:17]: Not a Etherscan explorer.

```suggestion
```

File: src/chains/definitions/meterTestnet.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    default: { name: 'MeterTestnetScan', url: 'https://scan-warringstakes.meter.io' },
+  },
+  contracts: {
+    
+  },
```
Comment (tmm) [inline src/chains/definitions/meterTestnet.ts:22]:
```suggestion
```

File: src/chains/index.ts [hunk 8]
```diff
@@ -67,6 +67,8 @@ export { mantle } from './definitions/mantle.js'
 export { mantleTestnet } from './definitions/mantleTestnet.js'
 export { metis } from './definitions/metis.js'
 export { metisGoerli } from './definitions/metisGoerli.js'
+export { meter } from './definitions/meter.js'
+export { meterTestnet } from './definitions/meterTestnet.js'
```
Comment (tmm) [inline src/chains/index.ts:71]: Alphabetize

```suggestion
export { meter } from './definitions/meter.js'
export { meterTestnet } from './definitions/meterTestnet.js'
export { metis } from './definitions/metis.js'
export { metisGoerli } from './definitions/metisGoerli.js'
```

**Review by tmm — CHANGES_REQUESTED** [review]
You should run `pnpm test src/chains/index.test.ts` to update the snapshot.

**Pull request merged**